src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php

```diff
@@ -271,8 +277,12 @@ public function visitGroupHeader(GroupHeader $groupHeader): void
         if ($groupHeader->getInitiatingPartyId() !== null && in_array($this->painFormat , array('pain.008.001.02','pain.008.003.02'))) {
             $newId = $this->createElement('Id');
             $orgId = $this->createElement('OrgId');
+            $schmeNm = $this->createElement('SchmeNm');
```

Comment:
Why are you adding this here? This is an optional element.

Reply:
I agree according to the specs it is, but our Spanish bank, I am unsure which bank it is will not accept the PAIN file without this information being present.

The moment we add this information all is well according to them.

Reply:
IMHO if it's specific to a single bank entity maybe you should patch the XML file after generating it, if my memory doesn't betray me I had to do that with Triodos because they required an additional element

Reply:
We could, but from the issues I read there is a user that has the exact same situation for an Irish bank. We know that the field is marked optional, and in this we we make sure that people can use it when required by their banks. Then why not give the people the option to add it?
This way there would be no need to alter an XML file that is already supposed to be a valid PAIN file.